pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -471,6 +471,18 @@ func (s *DiscoveryServer) WorkloadUpdate(id string, workloadLabels map[string]st
 	}
 
 	w.Labels = workloadLabels
+
+	// update workload labels, so that can improve perf of Proxy.SetWorkloadLabels
+	adsClientsMutex.RLock()
+	for _, connection := range adsClients {
+		// update node label
+		if connection.modelNode.IPAddresses[0] == id {
```

Comment:
don't we have a connections by IP map now?

Reply:
remembered, will do an optimization.

Reply:
Nope, i think the `connectionsByIP` is not right, try to think single control plane multi-network scenario.  Proxies from remote cluster may have the same ip address with a proxy in Local cluster.

Reply:
Nope, i think the `connectionsByIP` is not right, try to think single control plane multi-network scenario.  Proxies from remote cluster may have the same ip address with a proxy in Local cluster.